pkg/actions/nodegroup/upgrade.go

```diff
@@ -46,19 +50,34 @@ func (m *Manager) upgradeAndWait(nodeGroupName, version, launchTemplateVersion s
 		Version:       &version,
 	}
 
+	describeNodegroupOutput, err := m.ctl.Provider.EKS().DescribeNodegroup(&eks.DescribeNodegroupInput{
+		ClusterName:   &m.cfg.Metadata.Name,
+		NodegroupName: &nodeGroupName,
+	})
+
+	if err != nil {
+		return err
+	}
+
 	if launchTemplateVersion != "" {
-		describeNodegroupOutput, err := m.ctl.Provider.EKS().DescribeNodegroup(&eks.DescribeNodegroupInput{
-			ClusterName:   &m.cfg.Metadata.Name,
-			NodegroupName: &nodeGroupName,
-		})
-		if err != nil {
-			return err
+		lt := describeNodegroupOutput.Nodegroup.LaunchTemplate
+		if lt == nil || (lt.Id == nil && lt.Name == nil) {
+			return errors.New("cannot update launch template version because the nodegroup is not configured to use one")
 		}
-
 		input.LaunchTemplate = &eks.LaunchTemplateSpecification{
 			Version: &launchTemplateVersion,
 			Id:      describeNodegroupOutput.Nodegroup.LaunchTemplate.Id,
+			Name:    describeNodegroupOutput.Nodegroup.LaunchTemplate.Name,
```

Comment:
Just curious, why do we need to set `Name`?

Reply:
[one of the two need to be set](https://github.com/weaveworks/eksctl/pull/2953#discussion_r551967069), its unclear atm if this issue https://github.com/weaveworks/eksctl/issues/3102 is caused because the template only being referenced by the name in EKS, so I've added this as a precaution.